extensions/security-jpa/deployment/src/main/java/io/quarkus/security/jpa/deployment/QuarkusSecurityJpaProcessor.java

```diff
@@ -234,6 +250,32 @@ private ResultHandle lookupUserById(JpaSecurityDefinition jpaSecurityDefinition,
         return user;
     }
 
+    private static void activateCDIRequestContext(ClassCreator classCreator) {
+        try (MethodCreator methodCreator = classCreator.getMethodCreator("requireActiveCDIRequestContext",
+                DotName.createSimple(boolean.class.getName()).toString())) {
+            methodCreator.setModifiers(Modifier.PROTECTED);
+            methodCreator.returnBoolean(true);
+        }
+    }
+
+    private static boolean shouldActivateCDIReqCtx(HibernateOrmConfig hibernateOrmConfig,
+            SecurityJpaBuildTimeConfig secJpaConfig) {
+        var puConfig = hibernateOrmConfig.getAllPersistenceUnitConfigsAsMap().get(secJpaConfig.persistenceUnitName());
+        if (puConfig == null) {
+            throw new ConfigurationException("Persistence unit '" + secJpaConfig.persistenceUnitName()
+                    + "' specified with the 'quarkus.security-jpa.persistence-unit-name' configuration property"
+                    + " does not exist. Please set valid persistence unit name.");
+        }
```

Comment:
Note it's technically possible for the default persistence unit to not have any configuration, in which case the map wouldn't return any entry.

Are you sure you want to rely on `HibernateOrmConfig` directly? It seems dodgy. Wouldn't it make more sense to consume `io.quarkus.hibernate.orm.deployment.PersistenceUnitDescriptorBuildItem`? It has the advantage of being present regardless of configuration quirks, and exposes the multitenancy strategy through `getConfig().getMultiTenancyStrategy()`.

Reply:
> Note it's technically possible for the default persistence unit to not have any configuration, in which case the map wouldn't return any entry.

ouch, I'd guess that, it is not optional.`public HibernateOrmConfigPersistenceUnit defaultPersistenceUnit;` How is that possible? Anyway, thank you

> Are you sure you want to rely on HibernateOrmConfig directly? It seems dodgy. Wouldn't it make more sense to consume io.quarkus.hibernate.orm.deployment.PersistenceUnitDescriptorBuildItem? It has the advantage of being present regardless of configuration quirks, and exposes the multitenancy strategy through getConfig().getMultiTenancyStrategy().

I definitely want to use descriptor, thank you, I'll change it.

Reply:
> > Note it's technically possible for the default persistence unit to not have any configuration, in which case the map wouldn't return any entry.
> 
> ouch, I'd guess that, it is not optional.`public HibernateOrmConfigPersistenceUnit defaultPersistenceUnit;` How is that possible? Anyway, thank you

Hum, maybe I'm wrong... I'm sure I've seen this problem somewhere though. Anyway, even if it's not `null` right now, it definitely could be once we switch to `@ConfigMapping`. So, better stay away from this, and it seems you're working on it :)